anomalib/utils/callbacks/visualizer_callback.py

```diff
@@ -111,25 +111,26 @@ def on_test_batch_end(
             normalize = True  # raw anomaly maps. Still need to normalize
         threshold = pl_module.pixel_metrics.F1.threshold
 
-        for (filename, image, true_mask, anomaly_map) in zip(
-            outputs["image_path"], outputs["image"], outputs["mask"], outputs["anomaly_maps"]
-        ):
-            image = Denormalize()(image.cpu())
-            true_mask = true_mask.cpu().numpy()
-            anomaly_map = anomaly_map.cpu().numpy()
-
-            heat_map = superimpose_anomaly_map(anomaly_map, image, normalize=normalize)
-            pred_mask = compute_mask(anomaly_map, threshold)
-            vis_img = mark_boundaries(image, pred_mask, color=(1, 0, 0), mode="thick")
-
-            visualizer = Visualizer(num_rows=1, num_cols=5, figure_size=(12, 3))
-            visualizer.add_image(image=image, title="Image")
-            visualizer.add_image(image=true_mask, color_map="gray", title="Ground Truth")
-            visualizer.add_image(image=heat_map, title="Predicted Heat Map")
-            visualizer.add_image(image=pred_mask, color_map="gray", title="Predicted Mask")
-            visualizer.add_image(image=vis_img, title="Segmentation Result")
-            self._add_images(visualizer, pl_module, Path(filename))
-            visualizer.close()
+        if isinstance(outputs, dict) and "mask" in outputs.keys():
```

Comment:
It would be good if we could also visualize the results when no masks are not available, but we could address this separately.

Reply:
Exactly, similar to what @alexriedel1 proposed. A separate PR would address this